src/hash.cr

```diff
@@ -834,6 +842,10 @@ class Hash(K, V)
     entries_size == entries_capacity
   end
 
+  private def space_for_new_entry? : Bool
+    !entries_full?
+  end
+
```

Comment:
That seems pretty redundant.

Reply:
well, If you look at the definition it might look redundant, but here is an example of use:

```crystal
# If we have space to add a new entry ...
if !entries_full?
   ...
end
```
When I read `entries_full?` I have to "mental translate it" to _there is no more space for new entries_ and then `!entries_full?` should be read as _there is space for new entries_
And that's why the need of a comment `# If we have space to add a new entry ...` (because is much easier to read - we avoid the mental translation)

So instead of using a comment I just defined a more semantic method `space_for_new_entry?` which reads as:
```crystal
if space_for_new_entry?
   ...
end
```

Now, what I think it's redundant is that I leave the comment 🙃
```crystal
# If we have space to add a new entry ...
if space_for_new_entry?
...
end
```
I'll remove that comment. Let me know what do you think.

Reply:
Wouldn't be using `unless` makes this "problem" go away?

Reply:
Maybe I'm missing sth, but isn't it just the same, said differently? I mean _if sth is full_, then by definition there's no space for anything new.

Reply:
Well yes, by definition it is the same 🙂 it's just a matter of brain indirection. The way @ftarulla put it is direct, while thinking in terms of negations is a little bit more expensive. This said, to me both are good.

Reply:
Considering that the use case is in a private, low-level implementation, I don't see much point in extracting a method.
`if !entries_full?` is pretty specific. Might be changed to `unless entries_full?`, but that's still pretty similar.

Reply:
I'll resolve this conversation as it does not apply to the new implementation (although thx for the feedback on this subject!)
By the way, check out the new implementation! 😁